packages/runtime/src/schema.ts

```diff
@@ -102,6 +106,20 @@ const connectionArgsWithAccount = {
   },
 }
 
+const SetOptionsInput = new GraphQLInputObjectType({
+  name: 'SetOptionsInput',
+  fields: {
+    syncTimeout: {
+      type: GraphQLInt,
+      description: 'Maximum amount of time to lookup the stream over the network, in seconds',
+    },
+  },
+})
+
+type SetOptions = {
+  syncTimeout?: number
```

Comment:
Can we do this in a way that we don't need to update this for every new possible ceramic-side `CreateOpts` we want to expose?  Can we just expose a general purpose `createOpts` arg that just passes through to whatever `CreateOpts` the ceramic-level API exposes?

Reply:
I think it's better to make these objects ComposeDB-specific and opt-in to use Ceramic types as needed, for example the update options in ComposeDB are unrelated to the Ceramic ones.

Reply:
Yeah I disagree.  I think it's fine for ComposeDB to have its own specific options for creates and updates, but I think porting over every ceramic-level option one by one is going to be a maintenance nightmare, lead to a lot of extra work for us, and lead to unexpected feature gaps where things work in Ceramic but are missed in ComposeDB.  I think it would be fine if the ComposeDB-level Update struct had a single field like `CeramicUpdateOpts` that passed through the UpdateOpts from the Ceramic API, I don't think we need to flatten the options up into the top level.  In fact I think that separation could be nice to make it very clear where the options are that are coming from the Ceramic API vs what is ComposeDB-specific.

Reply:
With what you're suggesting we'd need to create the `CreateOpts` and `UpdateOpts` as GraphQL objects anyways, so to me this would be a bigger maintenance issue because that means any change on the Ceramic side would require an update on the ComposeDB side as well, we can't just "pass-through" options.

Reply:
gah, graphql always making things tricky!  What if we import the `CreateOpts` and `UpdateOpts` type definitions?  Can we then autogenerate the GraphQL objects off of those instead of having to manually re-define everything within them?  I just want to have a single place we can update and not have to try to manually keep the options in sync between js-ceramic and js-composedb

Reply:
Type informations are not available at runtime so I don't think that's an option

Reply:
gah, I don't know enough about how the GraphQL type generation works, so I'll just have to take your word on it that there's no way to do this.  That's really sad to me, I really hate having these arguments duplicated in multiple places and needing to always update ComposeDB any time an option is changed in js-ceramic.  But if that's how it is then I guess that's how it is.